src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -1,81 +1,88 @@
-import React, { FC, ReactNode, useState, useEffect } from "react";
+import { ReactNode, useState, useEffect } from "react";
 import styled from "@emotion/styled";
-import { number } from "prop-types";
+import axios from "axios";
 
-const exampleImages = [
-  "https://i.pinimg.com/564x/0f/71/0b/0f710bb81c2ff1aff8976239c18acfd2.jpg",
-  "https://i.pinimg.com/564x/57/37/e8/5737e8017b1c6946a6eb25b6db03a72e.jpg",
-  "https://i.pinimg.com/736x/55/ea/08/55ea0881688047362cb6d23f47166b65.jpg",
-  "https://i.pinimg.com/564x/48/40/05/4840054e287358f6b95e6dacfe395f2b.jpg",
-  "https://i.pinimg.com/564x/9c/82/a0/9c82a08a4a0e3b581122ab2576f1c07d.jpg",
-  "https://i.pinimg.com/564x/08/d1/67/08d167851acc6b5ca15785aab6825f23.jpg",
-  "https://i.pinimg.com/564x/4d/83/05/4d8305746c0a7fe3a5708ef300367aec.jpg",
-];
+const breakpoints = [576, 768, 992, 1200];
+const mq = breakpoints.map((bp) => `@media (min-width: ${bp}px)`);
+const accessKey = "kne3nTv7__ntufHd-qCFdSEZifJvlmPDyOVwha9jWqU";
+const unsplashUrl = `https://api.unsplash.com/photos/?client_id=${accessKey}`;
 
 interface ImageGridProps {
-  className?: string;
-  numberOfImgs?: number;
-  images?: Array<string>;
+  numberOfImgs: number;
+  imagesGridHeight: string;
+  imagesGridMaxWidth?: string;
   showModal?: boolean;
-  imagesGridWidth?: string;
-  imagesGridHeight?: string;
-  children?: ReactNode;
 }
 
-interface StyledComponentProps {
+interface StyledImageWrap {
   span?: number;
+  minWidth?: string;
   height?: string;
-  width?: string;
-  numberOfImgs?: number;
   row?: number;
   col?: number;
 }
 
+interface StyledImageGrid {
+  height: string;
+  numberOfImgs: number;
+  row: number;
+  col: number;
+  maxWidth?: string;
+}
+
 const ImageItem = styled.img`
   object-fit: cover;
   object-position: "center";
   min-width: 100%;
   height: 100%;
 `;
 
-let ImageWrap = styled("div")<StyledComponentProps>((props) => ({
+let ImageWrap = styled("div")<StyledImageWrap>((props) => ({
```

Comment:
here and below, please fix elsewhere in this file
```suggestion
let ImageWrap = styled.div<StyledImageWrap>((props) => ({
```